pkg/apis/workspaces/v1alpha1/commands.go

```diff
@@ -121,6 +121,14 @@ type ExecCommand struct {
 	// Optional list of environment variables that have to be set
 	// before running the command
 	Env []EnvVar `json:"env,omitempty"`
+
+	// +optional
+	// Whatever the command needs to be restarted when files changed.
+	// It should be set to "false" if command is capable to do automatic hotreload.
+	// Default value is "true" for group.kind: build, test.
+	// Default value is "false" for group.kind: run, debug.
+	// Default value is "false" for all commands without group.kind
+	RestartOnChange bool `json:"restartOnChange,omitempty"`
```

Comment:
If the default value is not fixed to `false` , then the Go definition should probably be a pointer (`*bool`) type:

```go
	RestartOnChange *bool `json:"restartOnChange,omitempty"`
```

This is the only way, in GO, to distinguish, for booleans, between:
- no value was provided (=> build a default value according to some logic)
- a `false` value.

(see [this comment](https://github.com/golang/go/issues/13284#issuecomment-158153815))

Reply:
This is a good point, but will also require nil checks wherever it's used.

Reply:
@davidfestal Good point.

I'm still considering if different default values based on command kind is a good idea.
Would it make sense for Che if we always default this to `true`?

Reply:
@kadel `true` means "no hot reload" => the user (or Che) will have to restart the command on each file change, right ?

Reply:
> @kadel `true` means "no hot reload" => the user (or Che) will have to restart the command on each file change, right ?

yes

Reply:
updated description with information that default value for `restartOnChange` is `true` (for all group.kind)